### PR TITLE
Chunked FieldUsageStatsResponse

### PR DESCRIPTION
**Pull request by original-brownbear**
These responses can become extremely large, chunk them.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/action/admin/indices/stats/FieldUsageStatsResponseTests.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xcontent.ToXContent;
+
+import java.util.List;
+import java.util.Map;
+
+public class FieldUsageStatsResponseTests extends ESTestCase {
+
+    public void testToXContentChunkPerIndex() {
+        final int indices = randomIntBetween(1, 100);
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/action/admin/indices/stats/FieldUsageStatsResponseTests.java:26]: May as well check this corner case too

```suggestion
        final int indices = randomIntBetween(0, 100);
```

File: server/src/test/java/org/elasticsearch/action/admin/indices/stats/FieldUsageStatsResponseTests.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            );
+        }
+        final FieldUsageStatsResponse response = new FieldUsageStatsResponse(indices, indices, 0, List.of(), perIndex);
+
+        final var iterator = response.toXContentChunked(ToXContent.EMPTY_PARAMS);
+        int chunks = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            chunks++;
+        }
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/action/admin/indices/stats/FieldUsageStatsResponseTests.java:53]: I suggest writing the chunks to an actual builder to verify they produce a well-formed document.

**Review by DaveCTurner — APPROVED** [review]
LGTM mod a couple of test comments

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**